### PR TITLE
Warn host and eventually close if inactive

### PR DESCRIPTION
**Pull request by GrantBartlett**
In relation to this: https://github.com/CnCNet/xna-cncnet-client/issues/88
Originally was this, but it was 2 years ago: https://github.com/CnCNet/xna-cncnet-client/pull/145

Checks the host hasn't been active for X seconds defined in the ClientDefinitions
Continues to check on past that time until X seconds until the hosted game is closed.

image

### CODE REVIEW

File: DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs [hunk 1]
```diff
@@ -245,6 +261,34 @@ private void MultiplayerName_RightClick(object sender, MultiplayerNameRightClick
...
+            );
+            hostInactiveWarningMessageBox.OKClickedAction = InactiveMessageBox_OKClicked;
+            hostInactiveWarningMessageBox.Show();
+        }
+
+        private void GameHostInactiveCheck_CloseInactiveGame(object sender, EventArgs e) => LeaveGameLobby();
+
+        private void InactiveMessageBox_OKClicked(XNAMessageBox messageBox) => gameHostInactiveCheck.HostIsAlive();
+
+        public override void OnMouseMove()
```
Comment (Metadorius) [inline DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs:282]: Wouldn't it be better to subscribe to an event if such exists?

File: ClientCore/ClientConfiguration.cs [hunk 2]
```diff
@@ -259,6 +259,14 @@ public string GetThemePath(string themeName)
 
         public string AllowedCustomGameModes => clientDefinitionsIni.GetStringValue(SETTINGS, "AllowedCustomGameModes", "Standard,Custom Map");
 
+        public string InactiveHostWarningTitle => clientDefinitionsIni.GetStringValue(SETTINGS, "InactiveHostWarningTitle", "Are you still here?");
+
+        public string InactiveHostWarningMessage => clientDefinitionsIni.GetStringValue(SETTINGS, "InactiveHostWarningMessage", "Your game may be closed due to inactivity.");
```
Comment (Metadorius) [inline ClientCore/ClientConfiguration.cs:264]: Better to leave as string literals in code and call L10N on them.

**Review by Metadorius — DISMISSED** [review]
Not exhaustive, just peeking at it between work xD

File: DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs (hunk 1 above)
Comment (GrantBartlett) [reply DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs:289]: > Wouldn't it be better to subscribe to an event if such exists?

don't follow this one sorry, could you give an example?

added the L10N change

Comment (Metadorius) [reply DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs:289]: I am not sure anymore, been too long lol

**Comment by GrantBartlett** [comment]
Any changes needed on this before it gets merged please? Thanks!

**Pull request closed without merging**